akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -314,33 +314,40 @@ object Behavior {
   /**
    * Execute the behavior with the given message
    */
-  def interpretMessage[T](behavior: Behavior[T], ctx: ActorContext[T], msg: T): Behavior[T] =
-    interpret(behavior, ctx, msg)
+  def interpretMessage[T](behavior: Behavior[T], ctx: ActorContext[T], msg: T): Behavior[T] = {
+    // optimization: duplicated logic with interpretSignal to avoid having to do the extra
+    // message typecheck in the happy case
+    behavior match {
+      case ext: ExtensibleBehavior[T] ⇒
+        val possiblyDeferredResult = ext.receive(ctx, msg.asInstanceOf[T])
+        start(possiblyDeferredResult, ctx)
+      case s: StoppedBehavior[T] ⇒ s
+      case EmptyBehavior         ⇒ UnhandledBehavior.asInstanceOf[Behavior[T]]
+      case IgnoreBehavior        ⇒ SameBehavior.asInstanceOf[Behavior[T]]
+      case _: UntypedPropsBehavior[_] ⇒
+        throw new IllegalArgumentException(s"cannot wrap behavior [$behavior] in " +
+          "Behaviors.setup, Behaviors.supervise or similar")
+      case d: DeferredBehavior[_] ⇒ throw new IllegalArgumentException(s"deferred [$d] should not be passed to interpreter")
+    }
```

Comment:
DeferredBehavior?

Reply:
There was two optimizations related to this match, one was the ordering to have the most likely one up top, the other to duplicate to avoid the extra message vs signal typecheck. Very strange that I somehow lost the same/undhandled clause though, I'll re-add that.